boards/FSM/Src/Io/Io_FlowMeters.c

```diff
@@ -0,0 +1,84 @@
+#include "main.h"
+#include "Io_SharedFrequencyOnlyPwmInput.h"
+#include "Io_FlowMeters.h"
+#include <assert.h>
+
+#define PRIMARY_FLOW_METER_CHANNEL HAL_TIM_ACTIVE_CHANNEL_1
+#define SECONDARY_FLOW_METER_CHANNEL HAL_TIM_ACTIVE_CHANNEL_2
+
+struct FlowMeters_Context
+{
+    struct FreqOnlyPwmInput *pwm_input;
+    size_t                   elapsed_tick;
+};
+
+static TIM_HandleTypeDef *       flow_meters_htim;
```

Comment:
I still think we should have one htim per flow meter. It's a coincidence that in this case that share the same timer. 

```
struct FlowMeter_Context {
  pwm_input
  elapsed_tick
  htim
}
```
Then write
```
    if (htim == primary_flow_meter.htim && htim->Channel == PRIMARY_FLOW_METER_CHANNEL)
    {
            Io_FlowMeters_TickPrimary();
     }

    // Repeat the above if-clause but for secondary
```

Reply:
fixed